### PR TITLE
#2841 - Fix Study Codes

### PR DESCRIPTION
**Pull request by katrina-aot-git**
Update SABC codes, and confirm other study codes.

Example:
image

### CODE REVIEW

**Comment by dheepak-aot** [comment]
A screenshot with program credential `undergraduate citation` and CIP `01.XXXX` will be helpful in PR description.

File: sources/packages/forms/src/form-definitions/educationprogram.json [hunk 1]
```diff
@@ -86,16 +98,37 @@
...
+                            "calculateValue": "value =\r\n{\r\n    11: \"Z\",\r\n    \"09\": \"Z\",\r\n    52: \"Z\",\r\n    \"01\": \"Y\",\r\n    \"03\": \"Y\",\r\n    \"05\": \"X\",\r\n    16: \"X\",\r\n    23: \"X\",\r\n    24: \"X\",\r\n    25: \"X\",\r\n    42: \"X\",\r\n    27: \"X\",\r\n    30: \"X\",\r\n    38: \"X\",\r\n    40: \"X\",\r\n    45: \"X\",\r\n    50: \"X\",\r\n    53: \"X\",\r\n    54: \"X\",\r\n    55: \"X\",\r\n    13: \"W\",\r\n    19: \"W\",\r\n    28: \"W\",\r\n    31: \"W\",\r\n    44: \"W\",\r\n    60: \"V\",\r\n    \"04\": \"U\",\r\n    10: \"U\",\r\n    14: \"U\",\r\n    15: \"U\",\r\n    29: \"U\",\r\n    41: \"U\",\r\n    26: \"T\",\r\n    22: \"S\",\r\n    43: \"S\",\r\n    51: \"R\",\r\n    61: \"R\",\r\n    39: \"Q\",\r\n    12: \"P\",\r\n    21: \"P\",\r\n    32: \"P\",\r\n    33: \"P\",\r\n    34: \"P\",\r\n    35: \"P\",\r\n    36: \"P\",\r\n    37: \"P\",\r\n    46: \"P\",\r\n    47: \"P\",\r\n    48: \"P\",\r\n    49: \"P\"\r\n};",
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/educationprogram.json:116]: Was there a need to update 03 and 26 ?? let me know if I am missing something here??

Comment (katrina-aot-git) [reply sources/packages/forms/src/form-definitions/educationprogram.json:116]: I went through all the codes in the linked spreadsheet and updated anything that did not match.

File: sources/packages/forms/src/form-definitions/educationprogram.json [hunk 2]
```diff
@@ -59,7 +59,8 @@
                     "key": "html3",
                     "type": "htmlelement",
                     "input": false,
-                    "tableView": false
+                    "tableView": false,
+                    "hideLabel": true
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/educationprogram.json:63]: The hide label being in the change seems to be weird. As the hide label is already enabled. 

image

Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:63]: I believe these new tags are part of the formio generation. I too tried to mimic and it seems to generate it.

File: sources/packages/forms/src/form-definitions/educationprogram.json (hunk 1 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:116]: @katrina-aot-git  can you please confirm with the business that these changes are needed to be synced? If yes please request them to add this as a part of the ticket.

**Review by guru-aot — COMMENTED** [review]
Nice work @katrina-aot-git . Please have a look on the comments.

File: sources/packages/forms/src/form-definitions/educationprogram.json (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:116]: The AC states to execute the check as @katrina-aot-git did.
If it is following the spreadsheet it is following the AC.

Comment (katrina-aot-git) [reply sources/packages/forms/src/form-definitions/educationprogram.json:116]: The acceptance criteria of the ticket was to find and fix the ones that did not match the spreadsheet. "Check all the Field of Study Codes and fix them accordingly to the [spreadsheet]".

File: sources/packages/forms/src/form-definitions/educationprogram.json (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:63]: @katrina-aot-git raised the concern to me also during development, it seems that it seems the editor causing it as an out-of-box behavior.
I was able to reproduce using the code from the main branch, just changing the order of the hidden fields and saving it.

**Review by andrewsignori-aot — APPROVED** [review]
Nice work and thanks for reviewing the spreadsheet. Looks good 👍 
Can you please add a comment to the ticket mentioning the extra differences found?

File: sources/packages/forms/src/form-definitions/educationprogram.json (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:116]: My bad I didn't notice the details in the AC.

**Review by dheepak-aot — APPROVED** [review]
Looks good @katrina-aot-git

**Review by guru-aot — APPROVED** [review]
LGTM, Nice work @katrina-aot-git

**Pull request merged**